### ISSUE TITLE
PS 1.7.4 product import problem with category-tree using same names

### ISSUE THREAD

**Issue by tiuz1** [PrestaShop/PrestaShop#12092]
Hello,

I always used csv to import my products at PS 1.6 but now I cannot import it correctly. It does not create the categories correctly and it also does not put the articles into the correct category.

I imported the following two articles and some categories are created (see screenshot).

[demo.txt](https://github.com/PrestaShop/PrestaShop/files/2741626/demo.txt)

![grafik](https://user-images.githubusercontent.com/2017880/50912075-8cd17100-1431-11e9-84b8-050ca27c2273.png)

But the Article (VERTIGO BLACK) should have been in the following three categories:
- Parfüm/Herren
- Parfüm/Herren/Aromatisch
- Parfüm/Herren/Aromatisch/Zitrus

And the Category _Parfüm/Damen/Blumig/Zitrus_ isn't even created.

**Comment by khouloudbelguith** [comment]
Hi @tiuz1,

I imported your file attached with PS1.7.5.0.
After the import, a product `VERTIGO (BLACK)` is created following these categories
![image](https://user-images.githubusercontent.com/16067358/50916338-2bae9b00-143b-11e9-9dfd-bee5d81cd835.png)
A product `DREAM ( GREEN )` is created following these categories
![image](https://user-images.githubusercontent.com/16067358/50916368-4719a600-143b-11e9-9d44-0d276aa6c3d1.png)

Thanks to check & feedback.

**Comment by tiuz1** [comment]
Do you need anything else to fix this problem?
Is there any way I can assist you?

**Comment by khouloudbelguith** [comment]
Hi @tiuz1,

In my case, I tried with your file csv but I did not manage to reproduce the issue, it is OK (See my previous screenshots)
The display is OK after the import in my screenshots?
Thanks!

**Comment by tiuz1** [comment]
In your example there is also a problem with the wrong category.
Vertigo Sport is listed in the Category Parfüm>Damen>Blumig>Zitrus, but it should be listed in the Herren (Men) Category (Parfüm/Herren/Aromatisch/Zitrus).

It seems like it does not travels down the trees, instead it looks for the last category name and put its int the first match, ignoring the tree structure.

**Comment by khouloudbelguith** [comment]
Hi @tiuz1,

Thanks for your feedback.
I tried with this file during the import
[demo_edited.txt](https://github.com/PrestaShop/PrestaShop/files/2808078/demo_edited.txt)
In the product page, they are OK
DREAM =>
![image](https://user-images.githubusercontent.com/16067358/51914816-f4982d80-23d9-11e9-8016-0b409cfcc55d.png)
VERTIGO =>
![image](https://user-images.githubusercontent.com/16067358/51914846-07126700-23da-11e9-8cf3-bbdbe98dd841.png)
Thanks to check & feedback.

**Comment by tiuz1** [comment]
Yes, if I try to trick the wrong import behaviour. But I cannot do this with hundreds or products.
But I want to the Main Category to be Parfüm/Herren/Aromatisch/Zitrus. So when I put it at the first position, the wrong category is assigned.
With PS 1.6 it worked flawlessly

**Comment by khouloudbelguith** [comment]
Hi @tiuz1,

after some tests with PS1.7.5.0 & PS1.7.4.2 & PS1.6.1.23 (Same behavior).
When I upload the file for the first time 
```
Product ID;Active (0/1);Name *;Categories (x,y,z...)
19075;1;DREAM;Parfüm/Damen/Blumig/Zitrus,Parfüm/Damen/Blumig,Parfüm/Damen
19747;1;VERTIGO;Parfüm/Herren/Aromatisch/Zitrus,Parfüm/Herren/Aromatisch,Parfüm/Herren
```
The associations for the second product (VERTIGO) is incorrect, 
It is displayed `Parfüm/Damen/Blumig/Zitrus` which is incorrect, it should be `Parfüm/Herren/Aromatisch/Zitru`
![image](https://user-images.githubusercontent.com/16067358/51969384-8f454a80-2474-11e9-935c-556fec842c63.png)
In the second time, I uploaded the same file with the same categories associated & it is OK.
![image](https://user-images.githubusercontent.com/16067358/51969471-c7e52400-2474-11e9-8344-57b7fd0ffebf.png)

So, this issue is reproduced only on the first time.
I attached a video record.
https://drive.google.com/file/d/19UJFUIbXbXdinPsUoHR3BxUkMtyM3_mv/view
I’ll add this to the debug roadmap so that it’s fixed. If you have already fixed it on your end or if you think you can do it, please do send us a pull request
Thanks!

**Comment by Hlavtox** [comment]
Closed, not relevant anymore.
